osu.Game/Screens/Edit/Compose/Components/Timeline/SamplePointPiece.cs

```diff
@@ -43,15 +49,35 @@ public SamplePointPiece(HitObject hitObject)
 
         protected override Color4 GetRepresentingColour(OsuColour colours) => AlternativeColor ? colours.Pink2 : colours.Pink1;
 
+        protected virtual double GetTime() => HitObject is IHasRepeats r ? HitObject.StartTime + r.Duration / r.SpanCount() / 2 : HitObject.StartTime;
```

Comment:
I don't have much faith in this not running into some weird floating point issue when requesting display, but let's see how it goes.

Reply:
It seems that it can run into issues if the sample point time is not calculated exactly the same way. I've added some leniency just in case.